docs/x509.rst

```diff
@@ -273,6 +289,73 @@ X.509 Certificate Object
 
         The dotted string value of the OID (e.g. ``"2.5.4.3"``)
 
+X.509 Extensions
+~~~~~~~~~~~~~~~~
+
+.. class:: Extensions
+
+    .. versionadded:: 0.9
+
+    An X.509 Extensions instance is an ordered list of extensions.  The object
+    is iterable to get every extension or you can reference it as a dictionary
+    using extension classes as the key.
+
+    .. doctest::
+
+        >>> cert.extensions[x509.BasicConstraints]
+        <BasicConstraints(ca=True, path_length=None, critical=True)>
```

Comment:
I think we should do one or the other on indexing or `get_extension_for_oid`, I don't really care which.